packages/core/src/StarknetWindowObject.ts

```diff
@@ -68,9 +222,29 @@ export type RpcMessage =
     }
   | {
       type: "wallet_switchStarknetChain"
-      params: SwitchStarknetChainParameter
+      params: SwitchStarknetChainParameters
       result: boolean
     }
+  | {
+      type: "starknet_addInvokeTransaction"
+      params: AddInvokeTransactionParameters
+      result: AddInvokeTransactionResult
+    }
+  | {
+      type: "starknet_addDeclareTransaction"
+      params: AddDeclareTransactionParameters
+      result: AddDeclareTransactionResult
+    }
+  | {
+      type: "starknet_addDeployAccountTransaction"
+      params: AddDeployAccountTransactionParameters
+      result: AddDeployAccountTransactionResult
+    }
+  | {
+      type: "starknet_signTypedData"
+      params: TypedData
+      result: string[]
+    }
```

Comment:
1. what's `SignTypedDataParameters`?
2. could u pls elaborate about `skipDeploy` - why is it needed, e.g. in what scenario you would auto-deploy an account due to off-chain signing? seems like a client-specific api

Reply:
Even if we don't want to support `skipDeploy`, I think it would be better to have

```{
      type: "starknet_signTypedData"
      params: { data: TypedData }
      result: string[]
    }
    ```
    
  This is more flexible and easy to extend in future.

Reply:
@avimak What about this?